sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/AbstractAsynchronousInstrument.java

```diff
@@ -88,6 +92,41 @@ LongResultSdk newResult(ActiveBatcher activeBatcher) {
       return new LongResultSdk(activeBatcher);
     }
 
+    @Override
+    public Observation observation(Long observation) {
+      return new LongObservationSdk(this.getActiveBatcher(), observation);
+    }
+
+    private static final class LongObservationSdk implements LongObservation {
```

Comment:
It bugs me that we need both this, and the LongResult, since they seem so very close to being exactly the same thing, just with the execution deferred in one case and immediate in the other.

Would it be simpler to just have the batch recorder keep a handle to the set of (instrument, value) pairs, then just make all the actual recordings at one time, rather than having to make all the wrapper classes? That is, could the Observation implementation just hold on to that Instrument and value?